junebug/tests/helpers.py

```diff
@@ -182,10 +182,19 @@ def start_server(self):
         port = reactor.listenTCP(
             0, Site(self.api.app.resource()),
             interface='127.0.0.1')
-        self.addCleanup(port.stopListening)
+        self.service._port = port
+        self.addCleanup(self.stop_server)
         addr = port.getHost()
         self.url = "http://%s:%s" % (addr.host, addr.port)
 
+    @inlineCallbacks
+    def stop_server(self):
+        yield self.service.stopService()
+        for service in self.service:
+            service.disownServiceParent()
+        for service in self.service.namedServices.values():
+            service.disownServiceParent()
```

Comment:
No, it seems like we still need it. I think that the first one just goes through unnamed services. You need to get the named services separately. I'm just not sure on the best way to get the named services.